common/djangoapps/util/log_sensitive.py

```diff
@@ -0,0 +1,175 @@
+"""
+Utilities for logging sensitive debug information such as authentication tokens.
+
+Usage:
+
+1. Read the warning in cli_gen_keys
+2. Generate keys using ``python3 -m common.djangoapps.util.log_sensitive gen-keys``
+   and follow the instructions it prints out
+3. When logging sensitive information, use like so::
+
+     logger.info(
+         "Received invalid auth token %s in Authorization header",
```

Comment:
Alternatively, always have the encrypted value start with a known prefix like `encrypted|`. This would aid in searching and not require an additional log message element to be independently specified.

It does seem useful to have that hint in the log output, not just in the source code.

Reply:
or `[encrypted: ZXI...fFo=|IYS...1KA==]` so that it's not part of the token per se, but formats well with the rest of the message.

Reply:
(This would require ignoring a trailing `]` in the decryption step in case it gets included by mistake.)

Reply:
Your suggested formatting works for me. I’m open to you just updating the sample log message, or changing return value. Ignoring the ] is up to you.

Reply:
Turns out `base64.b64decode` already ignores it!